content/03.when-to-use-dl.md

```diff
@@ -0,0 +1,51 @@
+Training such large models from scratch can be a costly and time-consuming undertaking [@arxiv:1906.02243].
+Luckily, most DL research in biology will not require nearly as much computation, and there are methods for reducing the amount of training required in some cases (described in [Tip 5](#architecture)).
+Specialized hardware such as discrete graphics processing units (GPUs) or custom DL accelerators can dramatically reduce the time and cost required to train models, but this hardware is not universally accessible.
+Currently, both GPU- and DL-optimized accelerator-equipped servers can be rented from cloud providers, though working with these servers adds additional cost and complexity.
+As DL becomes more popular, DL-optimized accelerators are likely to be more broadly available (for example, recent-generation iPhones already have such hardware).
+In contrast, traditional ML training can often be done on a laptop (or even a \$5 computer [@arxiv:1809.00238]) in seconds to minutes.
+
+Beyond the necessity for greater data and computational capacity in DL, building and training DL models generally requires more expertise than traditional ML models.
+Currently, there are several competing programming frameworks for DL such as Tensorflow [@arxiv:1603.04467] and PyTorch [@arxiv:1912.01703].
+These frameworks allow users to create and deploy entirely novel model architectures and are widely used in DL research as well as in indutry.
+This flexibility combined with the rapid development of the DL field has resulted in large, complex frameworks that can be daunting to new users.
+For readers new to software development but experienced in biology, gaining computational skills while interfacing with such complex industrial-grade tools can be a challenge.
+An advantage of ML over DL is that currently there are more tools capable of automating the model selection and training process.
+Automated ML (AutoML) tools such as TPOT [@doi:10.1007/978-3-319-31204-0_9], which is capable of using genetic programming to optimize ML pipelines, and Turi Create [@https://github.com/apple/turicreate], a task-oriented ML and DL framework which automatically tests multiple ML models when training, allow users to achieve competitive performance with only a few lines of code.
+Luckily, there are efforts underway to reduce the expertise required to build and use DL models.
+Indeed, both TPOT and Turi Create, as well as other tools such as AutoKeras [@arxiv:1806.10282], are capable of abstracting away much of the programming required for "standard" DL tasks.
+Projects such as Keras [@https://keras.io], a high-level interface for TensorFlow, make it relatively straightforward to design and test custom DL architectures.
+In the future, projects such as these are likely to bring DL experimentation within reach to even more researchers.
+
+There are some types of problems in which using DL is strongly indicated over ML.
```

Comment:
I think this part is missing "why" to synthesize the examples.

Reply:
Basically, what I'm getting at is that you don't need to start out with the most high powered tools there are now that there are more beginner-friendly ones that are stable. If you need a NN architecture that doesn't exist out of the box in one of the higher level tools but isn't so exotic that you _need_ TensorFlow primitives, a tool like Keras might be a good middle ground.